### PR TITLE
Consolidate write vs report methods

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Also, minor fix for reporting Total USD by utility bill scenario. (This is a fix to `report_runperiod_output_results`, not `write_runperiod_output_results`.)

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] Schematron validator (`EPvalidator.xml`) has been updated
- [ ] Sample files have been added/updated (via `tasks.rb`)
- [ ] Unit tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests`)
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected changes to simulation results of sample files

### CODE REVIEW

**Comment by shorowit** [comment]
@joseph-robertson It seems really weird to have separate methods for `write` vs `report`. They are supposed to write/report the same values, so they should have identical logic, yet their code is structured completely different. Why not just integrated the `runner.registerFoo` statements into the `write` method to ensure they are consistent and eliminate many lines of code?

**Comment by shorowit** [comment]
Looks like all of our reporting measure are like this. Yuck. Just seems like a recipe for bugs and inconsistency. Would you be open to consolidating all these methods?

**Comment by joseph-robertson** [comment]
Sure. (I wonder if we're missing some original motivation for separating them...?)

**Comment by joseph-robertson** [comment]
@shorowit I took a crack at consolidating methods in `ReportHPXMLOutput` and `ReportUtilityBills`. It looks less straightforward for `ReportSimulationOutput`; do you want to take that one?

File: ReportUtilityBills/measure.rb [hunk 1]
```diff
@@ -214,29 +214,57 @@ def run(runner, user_arguments)
...
-      results_out << ["#{bill_scenario_name}: #{key}: Marginal ($)", bill.annual_energy_charge.round(2)] if bill.annual_energy_charge != 0
-      results_out << ["#{bill_scenario_name}: #{key}: PV Credit ($)", bill.annual_production_credit.round(2)] if [FT::Elec].include?(key) && bill.annual_production_credit != 0
-      results_out << ["#{bill_scenario_name}: #{key}: Total ($)", bill.annual_total.round(2)] if bill.annual_total != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Fixed ($)", bill.annual_fixed_charge.round(2)] if bill.annual_fixed_charge != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Marginal ($)", bill.annual_energy_charge.round(2)] if bill.annual_energy_charge != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: PV Credit ($)", bill.annual_production_credit.round(2)] if [FT::Elec].include?(fuel_type) && bill.annual_production_credit != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Total ($)", bill.annual_total.round(2)] if bill.annual_total != 0
+
+      utility_bill_type_str = OpenStudio::toUnderscoreCase("#{bill_scenario_name} #{fuel_type} Fixed USD")
```
Comment (shorowit) [inline ReportUtilityBills/measure.rb:247]: Can't the code be simplified much further? After `results_out` is populated, loop through each item and do:
```
      name = OpenStudio::toUnderscoreCase(results_out_item[0])
      value = results_out_item[1]
      runner.registerValue(name, value)
      runner.registerInfo("Registering #{value} for #{name}.")
```
Wouldn't that work?

**Comment by shorowit** [comment]
@joseph-robertson What do you think about adding asserts to the reporting measure unit tests to check all of the output file rows were also registered on the runner?

**Comment by joseph-robertson** [comment]
I'll take a crack at consolidating methods in `ReportSimulationOutput`.

**Comment by shorowit** [comment]
It looks like all the reported variables are changing from "foo" to "foo_". For example, "fuel_use_electricity_total_m_btu" became "fuel_use_electricity_total_m_btu_". I just pushed a commit to fix this. I also changed the units from $ to USD in the utility bill measure to have consistency.

**Review by shorowit — APPROVED** [review]
Nice improvements here. Thanks @joseph-robertson.

**Pull request merged**